### ISSUE TITLE
False positive SIC_INNER_SHOULD_BE_STATIC_ANON when anonymous class references local field.

### ISSUE THREAD

**Issue by krallus** [spotbugs/spotbugs#2428]
The warning "Could be refactored into a named static inner class" [SIC_INNER_SHOULD_BE_STATIC_ANON](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sic-could-be-refactored-into-a-named-static-inner-class-sic-inner-should-be-static-anon) is shown for the following code (Java 8) even though the anonymous class references a local variable defined elsewhere in the method.
```java
...
final WebPageContentReader mockWebPageContentReader = mock(WebPageContentReader.class);

final ContentReadingService testContentReadingService = new ContentReadingService() {
	@Override
	protected SyndicationItemContentReader webPageContentReader() {
		return mockWebPageContentReader;
	}
};
...
```
This appears to be similar to an [old FindBugs problem](https://sourceforge.net/p/findbugs/bugs/873/).

**Comment by gtoison** [comment]
I tried reproducing the issue in the above test case (https://github.com/gtoison/spotbugs/commit/81a198a261e1cade8e6f20c80290a1e39fa9f972) but SIC_INNER_SHOULD_BE_STATIC_ANON is not reported.
Any idea what might be different in your case?

**Comment by krallus** [comment]
I am only getting back to this now and I also cannot reproduce it. There have been a number of updates and changes to both Spotbugs and my environment since then, so perhaps that's why it is no longer happening. In any case, I'm marking this as closed. Thank you for taking a look.

**Comment by krallus** [comment]
I am only getting back to this now and I also cannot reproduce it. There have been a number of updates and changes to both Spotbugs and my environment since then, so perhaps that's why it is no longer happening. In any case, I'm marking this as closed. Thank you for taking a look.

**Comment by gtoison** [comment]
Thank you for reverting, please do not hesitate to open an issue if the problem resurfaces!
